### PR TITLE
[PrivateKey]: add new C interface to get public key with cointype

### LINKED ISSUE #2656: [PrivateKey] Add shortcut method to get `PublicKey` from `CoinType`
We have these methods in `TWPrivateKey.h`: 
- `TWPrivateKeyGetPublicKeySecp256k1`
- `TWPrivateKeyGetPublicKeyEd25519`
- `TWPrivateKeyGetPublicKeyNist256p1`
- ...

Would be easier for the app to have a method like `TWPrivateKeyGetPublicKey(TWCoinType)`

### PR DESCRIPTION
**Pull request by Milerius**
## Description

fix #2656

## How to test

run c++ unit tests

## Checklist

- [x] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.

If you're adding a new blockchain

- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

**Review by catenocrypt — CHANGES_REQUESTED** [review]
I think the method to get public key type from a coin type should be in PrivateKey, and then there is no dependency issues.

**Comment by Milerius** [comment]
> I think the method to get public key type from a coin type should be in PrivateKey, and then there is no dependency issues.

That's the case: https://github.com/trustwallet/wallet-core/pull/2661/files#diff-7a19dd450838cf7c739919131c63557db3b685641ea91469fd00e287d4f564aeR73

**Review by hewigovens — APPROVED** [review]
One more Kotlin test would be nice

File: include/TrustWalletCore/TWCoinType.h [hunk 1]
```diff
@@ -11,13 +11,19 @@
...
+#include "TWPublicKeyType.h"
 
 TW_EXTERN_C_BEGIN
 
+/// Represents a private key.
+struct TWPrivateKey;
+
+/// Represents a public key.
+struct TWPublicKey;
+
```
Comment (catenocrypt) [inline include/TrustWalletCore/TWCoinType.h:26]:
```suggestion
```

This can be removed, not used

**Review by catenocrypt — APPROVED** [review]
I suggest to add these two pathways in a single test:

- A1: get pubkey for coin type (e.g. Eth)

- B1: get pubkey type for coin type
- B2: get pubkey for obtained pubkey type

Result should be the same.

**Review by hewigovens — APPROVED** [review]

**Comment by Milerius** [comment]
> I suggest to add these two pathways in a single test:
> 
> * A1: get pubkey for coin type (e.g. Eth)
> * B1: get pubkey type for coin type
> * B2: get pubkey for obtained pubkey type
> 
> Result should be the same.

Should we extend the C interface to have a `TWPrivateKeyGetPublicKeyByType(enum TWPublicKeyType pubkeyType)` or only write the second path in C++?

**Review by hewigovens — APPROVED** [review]

**Review by catenocrypt — APPROVED** [review]

**Review by miloserdow — APPROVED** [review]

**Pull request merged**